File: backend/src/main/java/axscalculator/axscalculator/models/mappool/ModBracketMap.java
package axscalculator.axscalculator.models.mappool;

public class ModBracketMap {
	private int beatmapId;
	private String beatmapName;
	private String beatmapUrl;
	private int modifier;
	private Gamemode gamemodeId;

	public ModBracketMap(int beatmapId, String beatmapName, String beatmapUrl, int modifier, Gamemode gamemodeId) {
		this.beatmapId = beatmapId;
		this.beatmapName = beatmapName;
		this.beatmapUrl = beatmapUrl;
		this.modifier = modifier;
		this.gamemodeId = gamemodeId;
	}

	public int getBeatmapId() {
		return beatmapId;
	}

	public String getBeatmapName() {
		return beatmapName;
	}

	public String getBeatmapUrl() {
		return beatmapUrl;
	}

	public int getModifier() {
		return modifier;
	}

	public Gamemode getGamemodeId() {
		return gamemodeId;
	}
}

File: src/app/models/osu-mappool/mod-bracket-map.ts
import { Gamemodes } from "../osu-models/osu-api";

export class ModBracketMap {
    invalid: boolean = false;
    beatmapId: number = null;
    beatmapName: string = null;
    beatmapUrl: string = null;
    modifier: number = null;
    gamemodeId: Gamemodes = Gamemodes.Osu;

    /**
     * Make a true copy of the given bracketmap
     * @param bracketMap the bracketmap
     */
    public static makeTrueCopy(bracketMap: ModBracketMap): ModBracketMap {
        const newModBracketMap = new ModBracketMap();

        newModBracketMap.invalid = bracketMap.invalid;
        newModBracketMap.beatmapId = bracketMap.beatmapId;
        newModBracketMap.beatmapName = bracketMap.beatmapName;
        newModBracketMap.beatmapUrl = bracketMap.beatmapUrl;
        newModBracketMap.modifier = bracketMap.modifier;
        newModBracketMap.gamemodeId = bracketMap.gamemodeId;

        return newModBracketMap;
    }
}
